google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -754,7 +754,9 @@ void beforeReadOrQueryLocked() {
     // Note that transactions are invalidated under some circumstances on the backend, but we
     // implement the check more strictly here to encourage coding to contract rather than the
     // implementation.
-    checkState(isValid, "Context has been invalidated by a new operation on the session");
+    if (!session.getIsMultiplexed()) {
+      checkState(isValid, "Context has been invalidated by a new operation on the session");
+    }
```

Comment:
This works here now, but I'm not sure this is the actual thing that needs to be changed to allow multiple operations on the same session. The `isValid` flag is set to false when `invalidate()` is called. That is called from `SessionImpl#setActive(SessionTransaction)` method. That again sets the `activeTransaction` on a `SessionImpl`. That field is what I think needs to be refactored for multiplexed sessions, as `SessionImpl#activeTransaction` is not really a thing anymore, as one `SessionImpl` can now have multiple transactions. At least; the latter is what I get from the fact that we just create a normal `SessionImpl` instance when we create a multiplexed sessions.